packages/headless/src/controllers/smart-snippet-suggestions/headless-smart-snippet-suggestions.ts

```diff
@@ -0,0 +1,95 @@
+import {Engine} from '../../app/headless-engine';
+import {buildController, Controller} from '../controller/headless-controller';
+import {search} from '../../app/reducers';
+import {loadReducerError} from '../../utils/errors';
+import {SearchSection} from '../../state/state-sections';
+
+export interface SmartSnippetSuggestionsDocumentId {
+  contentIdKey: string;
+  contentItValue: string;
+}
+
+/**
+ * The `SmartSnippetSuggestions` controller allows to manage additional queries for which a SmartSnippet model can provide relevant excerpts.
+ */
+export interface SmartSnippetSuggestions extends Controller {
+  /**
+   * The state of the SmartSnippet controller.
+   * */
+  state: SmartSnippetSuggestionsState;
+  /**
+   * Expand the specified snippet suggestion.
+   */
+  expand(identifier: SmartSnippetSuggestionsDocumentId): void;
+  /**
+   * Collapse the specified snippet suggestion.
+   */
+  collapse(identifier: SmartSnippetSuggestionsDocumentId): void;
+}
+
+/**
+ * A scoped and simplified part of the headless state that is relevant to the `SmartSnippet` controller.
+ */
+export interface SmartSnippetSuggestionsState {
+  /**
+   * The related questions for the current query
+   */
+  relatedQuestions: {
```

Comment:
The doc extraction tool can only extract one level deep, so we should adjust to:

```
export interface SmartSnippetSuggestionsState {
    relatedQuestions: <Interface>[]
}

Interface {
    ...
}
```

Relating to the naming comment above, if the controller is named RelatedQuestions or similar, we could shorted to just `questions`.